[CAMEL-14700] Improve colour contrast

aashnajena opened the pull request:
- drop-down submenus turn grey on hovering
- put an outline around the search bar to make it stand out
- added a grey shaded gradient (shadow) below nav-bar
- tried changing the background of the highlighted text to darker colours, but it was looking very odd, so I instead changed the colour of the text to camel orange - please comment on this. I have attached screenshots for both cases - background colour and orange text
- menu-items turn dark blue on hovering and while drop downs are open

![color_contast_latest](https://user-images.githubusercontent.com/32356795/76684458-36158780-6632-11ea-82a0-edda7200bd8b.png)

How the highlights look :

![camel_highlight_1](https://user-images.githubusercontent.com/32356795/76563287-20d31880-64cd-11ea-83be-2046b07c6bee.png)

And how the bold orange text looks :
![highlight_orange](https://user-images.githubusercontent.com/32356795/76563552-a951b900-64cd-11ea-9839-492ab4219765.png)
![highlight_orange_2](https://user-images.githubusercontent.com/32356795/76563559-ad7dd680-64cd-11ea-89b5-c89978ffc725.png)

aashnajena commented:
Hello @oscerd 
I'm so sorry, there weren't any comments by the other person on the JIRA issue so I wasn't aware that someone else was working on the same issue. I'll wait for the preview on PR 205, it seems like @avgupt has made some changes in the navbar colours.

zregvart commented:
Let's see about #205 first and come back to this, the preview is [here](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/index.html) to help compare the two approaches.

One odd thing I noticed is that I can no longer select text in this version, very odd. I tried in Firefox and Chromium.

aashnajena commented:
Aaah, I think I know where the problem is. Give me 5 minutes, I'll try to fix it!

aashnajena commented:
Are the rest of the changes okay? I've made changes such that text marked as <mark> are displayed boldly in orange, but user-selected text becomes white with dark orange background. Please check now! 
![user-selection](https://user-images.githubusercontent.com/32356795/76612681-ce344380-6542-11ea-8cb6-d71cead0aa0e.png)

aashnajena commented:
Hi I've added a change - menu items now turn dark blue on hovering (earlier there has no visible change on hovering, which makes it less accessible). Updated the first screenshot to reflect this!

zregvart commented:
Preview is [here](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/index.html)

aashnajena commented:
Hey! I don't remember exactly which blogs I read, but what most blogs said is that people with low vision have problem observing cursor changes, since the cursor is often very small, even with magnified screens. Hence in order to know what they are selecting (or what they are hovering on), it is crucial to have contrasting colours (like blue on white). Because even if they can't read the entire menu properly, they should be able to read the part they are selecting/hovering on. In the current website, hovering on the main menu just changes the cursor, the focused element is not decorated at all. This link might be helpful

https://ux.stackexchange.com/questions/126893/does-hover-state-button-text-need-to-meet-accessibility-guidelines

And I've followed this guide to understand how to make accessible menus :
https://www.smashingmagazine.com/2017/11/building-accessible-menu-systems/

aashnajena commented:
About the <mark> colour, maybe we can have a third person's opinion! I think the colour is very close to the background colour. I honestly cannot make out the difference in colours if I don't have my spectacles on! But yes, as soon as I try to make that colour any darker, it starts looking odd, which is why I resorted to colouring the text instead of the background. 

Besides, I went to several webpages and tried to select text. In all the cases, the background colour of the highlighted text is dark orange or dark blue - even though it's not very aesthetic.

zregvart commented:
Anyone have any thoughts on this? I'm fully accepting that I'm wrong on this :)

I guess the change can be seen on the frontpage ([before](https://camel.apache.org/), [after](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/index.html)), look at the highlighted text and the menu colour change on hover, 

In the documentation ([before](https://camel.apache.org/manual/latest/), [after](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/manual/latest/getting-started.html))
On the release page ([before](https://camel.apache.org/releases/release-3.1.0/), [after](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/releases/release-3.1.0/))


cc @oscerd @djencks @delawen

oscerd commented:
+1 for me

aashnajena commented:
Hey, the hovering on sub-menu items looks different for on my localhost and the preview provided above. This is how it looks for me - 
![menu-selection](https://user-images.githubusercontent.com/32356795/77634149-823bc280-6f76-11ea-89b5-472df5c0623e.png)

aashnajena commented:
Hi, does the sub-menu selection look like this for you on the preview? I just tried pulling changes from master and rebasing, my localhost still shows this :
![menu-selection](https://user-images.githubusercontent.com/32356795/77636179-b49aef00-6f79-11ea-9022-363f3365e7a8.png)

zregvart commented:
This is how it looks for me on the last [preview](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/index.html).

![Screenshot_2020-03-26 Home - Apache Camel](https://user-images.githubusercontent.com/1306050/77663133-03965380-6f7d-11ea-92ff-8bba7f2bc44a.png)

Not sure if that was intended, I think if we're going to have the highlight, it should go end to end.

I guess we also broke the positioning of the menu at some point. The menus used to start at the base grey line below separating the main navigation and the content, now it overlaps.

aashnajena commented:
I don't see this happening on my localhost. I just pulled changes and rebased again to make sure it's not happening because of some other merged PRs. Any idea why there is this discrepancy?

aashnajena commented:
> This is how it looks for me on the last [preview](https://builds.apache.org/job/Camel.website/job/PR-207/Preview/index.html).
> 
> ![Screenshot_2020-03-26 Home - Apache Camel](https://user-images.githubusercontent.com/1306050/77663133-03965380-6f7d-11ea-92ff-8bba7f2bc44a.png)
> 
> Not sure if that was intended, I think if we're going to have the highlight, it should go end to end.
> 
> I guess we also broke the positioning of the menu at some point. The menus used to start at the base grey line below separating the main navigation and the content, now it overlaps.

Hi can you check the current website once? For me, the sub-menu seems broken there also, it's not starting from below the line. I tried on 3 different browsers. I'm trying to resolve this in a separate PR.

![broken-menu](https://user-images.githubusercontent.com/32356795/77819262-8a733980-70ff-11ea-92de-8b9495986d9e.png)

aashnajena commented:
Hey, I have resolved this issue in #282 and tested it here, the highlight becomes end to end and the sub-menu starts where required. Please review it!

aashnajena commented:
@zregvart can you please tell me what is the verdict on this PR?
